internal/validation/suite.go

```diff
@@ -11,25 +10,23 @@ import (
 )
 
 type ProviderConfig struct {
-	ProviderName  string
-	EnvVarName    string
-	ClientFactory func(apiKey string) v1.CloudClient
+	ProviderName string
```

Comment:
I don't think we need provider name since we can get provider id from the credential. Lets still keep the ProviderConfig in case we need to add test suit specific attributes,